homeassistant/components/wallbox/__init__.py

```diff
@@ -5,41 +5,49 @@
 import requests
 from wallbox import Wallbox
 
-from homeassistant import exceptions
-from homeassistant.config_entries import ConfigEntry
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
 from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, HomeAssistantError
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
 from .const import CONF_CONNECTIONS, CONF_ROUND, CONF_SENSOR_TYPES, CONF_STATION, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORMS = ["sensor"]
+PLATFORMS = ["sensor", "number"]
 UPDATE_INTERVAL = 30
 
 
-class WallboxHub:
+class WallboxHub(DataUpdateCoordinator):
     """Wallbox Hub class."""
 
-    def __init__(self, station, username, password, hass):
+    def __init__(
+        self, station, username: str, password: str, hass: HomeAssistant
+    ) -> None:
         """Initialize."""
         self._station = station
-        self._username = username
-        self._password = password
-        self._wallbox = Wallbox(self._username, self._password)
+        self._wallbox = Wallbox(username, password)
```

Comment:
Why co you need line31 ? hass is automatically added as first parameters to all HA calls.

Reply:
If I leave it out (and add it as a parameter to class functions), I seem to (at least) be unable to access hass in the _async_update_data function. I get the following warning: wallbox integration not ready yet: _async_update_data() missing 1 required positional argument: 'hass'; Retrying in background.

Any tips?

Reply:
Did a couple of tests, and seems the data_coordinator do not use hass. So your change is ok.